Update Collection Requirements (#195)
ansible-collections/overview: collection_requirements.rst

******************************************

> Ansible Community Collections Requirements

Reviewer: To emphasize that this is not for Community Collections, but for collections included in the Ansible Community Package:
```suggestion
Ansible Community Package Collections Requirements
```

Author: Sounds good, committed, thanks!

.. note::

  `Inclusion of a new collection <https://github.com/ansible-collections/ansible-inclusion>`_ in the Ansible package is ultimately at the discretion of the `Ansible Community Steering Committee <https://github.com/ansible/community-docs/blob/main/ansible_community_steering_committee.rst>`_. Every rejected candidate will get feedback. Differences of opinion should be taken to a dedicated `Community Topic <https://github.com/ansible-community/community-topics/issues>`_ for discussion and a final vote.

Feedback
========

As with any project it's very important that we get feedback from users, contributors and maintainers. We recognize that the move to Collections is a big change in how Ansible is developed and delivered.
As with any project it's very important that we get feedback from users, contributors, and maintainers. We recognize that the move to Collections is a big change in how Ansible is developed and delivered.

Please raise feedback by:

> * Discussing in the ``#ansible-community`` `libera.chat <https://libera.chat/>`_ IRC channel.

Author: Changed, PTAL

> * The `news-for-maintainers repository <https://github.com/ansible-collections/news-for-maintainers>`_ to track changes that Collection maintainers should be aware of.

Reviewer: ```suggestion
* The `news-for-maintainers repository <https://github.com/ansible-collections/news-for-maintainers>`_ to track changes that Collection maintainers should be aware of. Subscribe only to issues if you want less traffic.
```

Author: makes sense to mention:)

* The `Bullhorn <https://github.com/ansible/community/wiki/News#the-bullhorn>`_ Ansible contributor newsletter.

Collection Infrastructure
=========================

> Community collections

Reviewer: Is this about `community.*` collections, or about collections included in the Ansible package? The below rules are about the latter, but this sounds like the former. How about:
```suggestion
Collections included in the Ansible community package
```

Author: Works for me:)

* MUST have a Code of Conduct (hereinafter the CoC).

  * The collection's CoC MUST be compatible with the `Ansible Code of Conduct <https://docs.ansible.com/ansible/latest/community/code_of_conduct.html>`_.
  * The Collections SHOULD consider using the Ansible CoC if they do not have a CoC that they consider better.
  * The `Diversity and Inclusion working group <https://docs.ansible.com/ansible/latest/community/communication.html#working-groups>` may evaluate all CoCs and object to a collection's inclusion based on the CoCs contents
  * The CoC MUST be linked from the ``README.md`` file, or MUST be present or linked from the ``CODE_OF_CONDUCT.md`` file in the collection root.
  
* MUST be published to `Ansible Galaxy <https://galaxy.ansible.com>`_.

Python Compatibility
====================

A collection MUST be developed and tested using the below Python requirements as Ansible supports a wide variety of machines.

The collection should adhere to the tips mentioned in the official `Ansible Development Guide <https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3>`_.

Python Requirements
-------------------

> Python requirements for a collection vary between ``controller-environment`` and ``other-environment``. On the controller-environment, the Python versions required may be higher than what is required on the other-environment. While developing a collection, you need to understand the definitions of both  controller-environment and other-environment to help you choose Python versions accordingly: 

Reviewer: Or maybe also single `*`. But double backticks is the wrong formatting IMO, except if we always use teletype for these words (which we do not).
```suggestion
Python requirements for a collection vary between **controller-environment** and **other-environment**. On the controller-environment, the Python versions required may be higher than what is required on the other-environment. While developing a collection, you need to understand the definitions of both  controller-environment and other-environment to help you choose Python versions accordingly: 
```

* ``controller-environment``: The plugins/modules always run in the same environment (Python interpreter, venv, host, etc) as ansible-core itself.

> * ``other-environment``: It is possible, even if uncommon in practice, for the plugins/modules to run in a different environment than ansible-core itself.

Reviewer: Same here.

Author: i think the bullet points highlight them enough without any additional formatting, so I'll just remove it

> One example scenario where the ``even if`` clause comes into play is when using Cloud modules. These modules mostly run on the controller node but in some environments, the controller might run on one machine inside a demilitarized zone which cannot directly access the cloud machines. The user has to have the cloud modules run on a bastion host/jump server which has access to the cloud machines.

Reviewer: ```suggestion
One example scenario where the "even if" clause comes into play is when using Cloud modules. These modules mostly run on the controller node but in some environments, the controller might run on one machine inside a demilitarized zone which cannot directly access the cloud machines. The user has to have the cloud modules run on a bastion host/jump server which has access to the cloud machines.
```

**************************************************
Ansible Community Package Collections Requirements
**************************************************

.. contents:: Topics

Overview
========

> This document is for maintainers of Ansible Community collections, living in the `ansible-collections <https://github.com/ansible-collections>`_ repository or included in the Ansible community package (hereinafter the Collections), to provide them help, advice, and guidance on making sure their collections are correct.

Reviewer: ```suggestion
This document is for maintainers of Ansible Community collections, living in the `ansible-collections <https://github.com/ansible-collections>`_ repository or included in the Ansible community package (hereinafter the collections), to provide them help, advice, and guidance on making sure their collections are correct.
```
Do we really have to use a capital C here? If we do, there are a LOT of more places to check and possibly adjust (I saw a few which would need a capital C as well). I think it's better to stick to the lower-case form.

Author: yep, let's use it in lower-case

Author: changed everywhere, thanks!

.. note::

  `Inclusion of a new collection <https://github.com/ansible-collections/ansible-inclusion>`_ in the Ansible package is ultimately at the discretion of the `Ansible Community Steering Committee <https://github.com/ansible/community-docs/blob/main/ansible_community_steering_committee.rst>`_. Every rejected candidate will get feedback. Differences of opinion should be taken to a dedicated `Community Topic <https://github.com/ansible-community/community-topics/issues>`_ for discussion and a final vote.

Feedback
========

As with any project it's very important that we get feedback from users, contributors, and maintainers. We recognize that the move to Collections is a big change in how Ansible is developed and delivered.

Please raise feedback by:

> * Discussing in the `#community:ansible.com <https://matrix.to/#/#community:ansible.com>`_ Matrix channel bridget with the ``#ansible-community`` channel on Libera.Chat IRC. See the `Ansible Communication Guide <https://docs.ansible.com/ansible/latest/community/communication.html#real-time-chat>`_ for details.

Reviewer: ```suggestion
* Discussing in the `#community:ansible.com Matrix room <https://matrix.to/#/#community:ansible.com>`_ bridges with the ``#ansible-community`` channel on Libera.Chat IRC. See the `Ansible Communication Guide <https://docs.ansible.com/ansible/latest/community/communication.html#real-time-chat>`_ for details.
```

All modules and plugins MUST:

* Include a `DOCUMENTATION <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-block>`_ block.
* Include an `EXAMPLES <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#examples-block>`_ block (except where not relevant for the plugin type).

> * Use FQCNs when referring to modules, plugins and documentation fragments inside and outside the collection (including ``ansible.builtin.`` for the listed entities from `Ansible-base <https://docs.ansible.com/ansible/devel/collections/ansible/builtin/>`_).

Reviewer: ```suggestion
* Use FQCNs when referring to modules, plugins and documentation fragments inside and outside the collection (including ``ansible.builtin.`` for the listed entities from `Ansible-core <https://docs.ansible.com/ansible/devel/collections/ansible/builtin/>`_).
```

Collections are required to include a changelog.  To give a consistent feel for changelogs across collections and ensure changelogs exist for collections included in the ``ansible`` package we suggest you use `antsibull-changelog <https://github.com/ansible-community/antsibull-changelog>`_ to maintain and generate this but other options exist.  Preferred (in descending order):

#. Use antsibull-changelog (preferred).

> #. Provide ``changelogs/changelog.yaml`` in the `correct format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_.

Reviewer: ```suggestion
#. Provide ``changelogs/changelog.yaml`` in the `correct format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_. (You can use ``antsibull-lint changelog-yaml /path/to/changelog.yaml`` to validate the format.)
```

These guidelines are the policy for inclusion in the Ansible package and are in addition to any
licensing and legal concerns that may otherwise affect your code.

Repository management
=====================

> Every Community collection MUST have a public SCM repository, and releases of the collection MUST be tagged in this repository.

Reviewer: ```suggestion
Every collection MUST have a public SCM repository, and releases of the collection MUST be tagged in this repository.
```
Same here. This applies to **all** collections included in Ansible, not only to `community.*` collections.

Author: `Community` is not about `community.*` collections. Its aim is to highlight that this is not about collections living on Automation Hub

Author: Solved

CI Testing
===========

> 

Author: OK, will think of this a bit later

> 

Author: added, PTAL

  * All entries in ignores.txt MUST have a justification in a comment in the ignore.txt file for each entry.  For example ``plugins/modules/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path``.
  * Reviewers can block acceptance of a new collection if they don't agree with the ignores.txt entries.

> * You MUST run CI against each of the ``major versions`` (2.10, 2.11, 2.12, etc) of ``ansible-base``/``ansible-core`` that the collection supports. (Usually the ``HEAD`` of the stable-xxx branches.)

Reviewer: Double backticks is not a substitute for quotation marks.
```suggestion
* You MUST run CI against each of the "major versions" (2.10, 2.11, 2.12, etc) of ``ansible-base``/``ansible-core`` that the collection supports. (Usually the ``HEAD`` of the stable-xxx branches.)
```

Other things
============

> * ansible-core's runtime.yml

Author: ```suggestion
* ansible-core's runtime.yml
```
Can we remove this line?

Reviewer: I think so, since changes to it are no longer accepted AFAIK.

Author: Removed, thanks
